### PR TITLE
.github/workflows/doc-build.yml: Fix live doc building

### PR DESCRIPTION
**Pull request by kwankyu**
Fixes the problem seen in

https://github.com/sagemath/sage/actions/runs/6761900181/job/18377280664

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
@mkoeppe

**Comment by kwankyu** [comment]
Building live doc failed on push to develop. I don't understand the failure:
```
2023-11-05T17:35:23.7872718Z [sagemath_doc_html-none] [polynomia] building [inventory]: targets for 65 source files that are out of date
2023-11-05T17:35:23.7874171Z [sagemath_doc_html-none] [polynomia] updating environment: [new config] 65 added, 0 changed, 0 removed
2023-11-05T17:35:23.7876265Z [sagemath_doc_html-none] [polynomia] WARNING: error while formatting arguments for sage.rings.polynomial.laurent_polynomial.LaurentPolynomial.dict: The given source does not contain 'def'
```
from https://pipelinesghubeus13.actions.githubusercontent.com/6JnDqfSihQKOyEkfoNVGbkdBAYv1aMk1pe4vbKvQiAtlTRKBXz/_apis/pipelines/1/runs/44130/signedlogcontent/11?urlExpires=2023-11-06T01%3A47%3A12.2695522Z&urlSigningMethod=HMACV1&urlSignature=xiSqDibZ3%2BvAFRX0osK65fXCd9JkFK90wDoynshaENg%3D

Any clue?

**Comment by kwankyu** [comment]
This is part of `doc-build.yml`:
```bash
          echo '</head>' >> ./docs/CHANGES.html
          echo '<body>' >> ./docs/CHANGES.html
          (cd /sage/local/share/doc/sage/html && git diff HEAD^ -- *.html; rm -rf .git) > ./docs/diff.txt
          /sage/sage -python - << EOF
          import re, html
          with open('./docs/diff.txt', 'r') as f:
              diff_text = f.read()
              diff_blocks = re.split(r'^(?=diff --git)', diff_text, flags=re.MULTILINE)
              out_blocks = []
              for block in diff_blocks:
                  match = re.search(r'^diff --git a/(.*) b/\1', block, flags=re.MULTILINE)
                  if match:
                      path = 'html/' + match.group(1)
                      out_blocks.append(f'<p class="diff"><a href="{path}">{path}</a>&emsp;</p>\n<pre><code class="language-diff">' + html.escape(block).strip() + '</code></pre>')
              output_text = '\n'.join(out_blocks)
          with open('./docs/diff.html', 'w') as f:
              f.write(output_text)
          EOF
          cat ./docs/diff.html >> ./docs/CHANGES.html
          echo '</body>' >> ./docs/CHANGES.html
          echo '</html>' >>./docs/CHANGES.html
          rm ./docs/diff.txt ./docs/diff.html
          (cd /sage/local/share/doc/sage/html && git reset --hard HEAD)
          # For some reason the deploy step below cannot find /sage/...
          # So copy everything from there to local folder
          # We also need to replace the symlinks because netlify is not following them
          cp -r -L /sage/local/share/doc/sage/html ./docs
          cp  /sage/local/share/doc/sage/index.html ./docs
          # Zip everything for increased performance
          zip -r docs.zip docs  
```
In the line
```
          (cd /sage/local/share/doc/sage/html && git diff HEAD^ -- *.html; rm -rf .git) > ./docs/diff.txt
```
`.git` directory is removed. Then
```
          (cd /sage/local/share/doc/sage/html && git reset --hard HEAD)
```
it `git reset --hard HEAD`. This looks suspicious. `.git` directory is already gone. So what `git reset ...` does?

**Comment by kwankyu** [comment]
I removed the suspicious lines. Let's see.

**Comment by kwankyu** [comment]
Embarrassingly it is me who put the erroneous line in #36483!

**Comment by kwankyu** [comment]
The patch works well on my own test repo: https://github.com/kwankyu/sage/actions/runs/6766229169/job/18387044825

Hopefully fixed the root cause of the failure. We will see in the next release.

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -162,7 +162,7 @@ jobs:
           EOF
           echo '</head>' >> ./docs/CHANGES.html
           echo '<body>' >> ./docs/CHANGES.html
-          (cd /sage/local/share/doc/sage/html && git diff HEAD^ -- *.html; rm -rf .git) > ./docs/diff.txt
```
Comment (mkoeppe) [inline .github/workflows/doc-build.yml]: The purpose of removing the .git directory is that it does not end up in `docs.zip`

Comment (kwankyu) [reply .github/workflows/doc-build.yml]: Sure. I was stupid.

**Review by mkoeppe — APPROVED** [review]
LGTM

**Comment by kwankyu** [comment]
Thanks!

**Pull request merged**